src/index.js

```diff
@@ -92,7 +80,65 @@ export default class AutoCheckElement extends HTMLElement {
   }
 }
 
-function check(autoCheckElement: AutoCheckElement) {
+function makeDeferred<T>(): [Promise<T>, (T) => Promise<T>, (Error) => T] {
+  let resolve
+  let reject
+  const promise = new Promise(function(_resolve, _reject) {
+    resolve = _resolve
+    reject = _reject
+  })
+
+  if (!resolve) throw new Error('invariant: resolve')
+  if (!reject) throw new Error('invariant: reject')
+
+  return [promise, resolve, reject]
+}
+
+function makeAbortController() {
+  if ('AbortController' in window) {
+    return new AbortController()
+  }
+  return {signal: null, abort() {}}
+}
+
+async function slidingPromiseFetch(el: HTMLElement, url: string, options: RequestOptions = {}): Promise<Response> {
+  let request = requests.get(el)
+  const [promise, resolve, reject] = makeDeferred<Response>()
```

Comment:
While we're refactoring, is it worth removing `makeDeferred` which is a bit of an anti pattern? This is the only call-site for the function, and this function itself aways returns the Promise, so it is probably more readable to `return new Promise(() => { })` here, and then you have `resolve`/`reject` in scope.

Reply:
Could you elaborate? Other than inlining the function body here I'm not sure how I would go about this. Inlining the function body is less readable to me 👀